lua/lspconfig/ltex.lua

```diff
@@ -4,24 +4,13 @@ local util = require 'lspconfig/util'
 configs.ltex = {
   default_config = {
     cmd = { 'ltex-ls' },
-    filetypes = { 'tex', 'bib', 'markdown' },
+    filetypes = { 'plaintex', 'tex', 'bib', 'markdown' },
     root_dir = util.find_git_ancestor,
-    settings = {
-      ltex = {
-        enabled = { 'latex', 'tex', 'bib', 'markdown' },
-        checkFrequency = 'edit',
-        language = 'en',
-        diagnosticSeverity = 'information',
-        setenceCacheSize = 2000,
-        additionalRules = {
-          enablePickyRules = true,
-          motherTongue = 'en',
-        },
-        dictionary = {},
-        disabledRules = {},
-        hiddenFalsePositives = {},
-      },
-    },
+    get_language_id = function(_, filetype)
+      if vim.tbl_contains({ 'tex', 'plaintex' }, filetype) then
+        return 'latex'
+      end
```

Comment:
Mapping plain T<sub>E</sub>X to `latex` is against the [language IDs recommended by the LSP spec](https://microsoft.github.io/language-server-protocol/specification#textDocumentItem), and it's a bit of a hack. *If* you want enable users to check plain T<sub>E</sub>X as well, then send `tex` and not `latex` (it would also be OK to send `plaintex`, but then LT<sub>E</sub>X LS needs to be changed first). Users then need to add `tex` (or `plaintex`) to [`ltex.enabled`](https://valentjn.github.io/ltex/settings.html#ltexenabled).

(The reason LT<sub>E</sub>X is not enabled for `tex` by default is that most plain T<sub>E</sub>X documents are packages, class files, etc., which shouldn't be checked.)

Reply:
Note that Vim treats both TeX and LaTeX as `filetype=tex` -- these are _Vim's_ internal filetypes, not LSP language IDs.

Reply:
There's also `context`.

Reply:
isn't bib the filetype for bibtex? So we should remap this to bibtex too right?

Reply:
Yes, the filetype for BibTeX is (only!) `bib`.

Reply:
I went ahead and mapped plaintex -> tex, bib -> bibtex, and tex -> latex which should be correct with my understanding of vim's filetype settings

Reply:
I went ahead and also added rst and xhtml support as those filetype plugins are also built into neovim, re: org/rsweave would need to ask users to define an autocommand as there are not available plugins

Reply:
you could map `rnoweb` to Sweave

Reply:
I did :)